src/seedu/addressbook/commands/AddCommand.java

```diff
@@ -35,16 +35,24 @@ public AddCommand(String name,
                       String phone, boolean isPhonePrivate,
                       String email, boolean isEmailPrivate,
                       String address, boolean isAddressPrivate,
+                      
                       Set<String> tags) throws IllegalValueException {
         final Set<Tag> tagSet = new HashSet<>();
+        String[] fullAddress = address.split(",");
+
+        String blockName = fullAddress[0];
+        String streetName = fullAddress[1];
+        String unitName = fullAddress[2];
+        String postalName = fullAddress[3];
+
         for (String tagName : tags) {
             tagSet.add(new Tag(tagName));
         }
         this.toAdd = new Person(
                 new Name(name),
```

Comment:
would it be better if you split the address in the `Address` class than here? think why or why not?